### PR TITLE
feat(auth): implement dependency for authenticating users

### PR DESCRIPTION
**Pull request by jankrokos**

### CODE REVIEW

File: rms/__main__.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 
 
 @app.get("/health")
-def health_endpoint():
-    return {"healthy": "true"}
+def health_endpoint(user: User = Depends(get_current_user)):
+    return {
+        "healthy": "true",
+        "user": user
+    }
```
Comment (CreativePhilip) [inline rms/__main__.py:21]: leftover

File: rms/auth/dependencies.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from datetime import datetime
+
+from fastapi import Request, HTTPException, status
+
+from rms.auth.managers import UserCookieManager, UserManager
+
+
+def get_current_user(request: Request):
```
Comment (CreativePhilip) [inline rms/auth/dependencies.py:8]: missing return type annotation

**Review by jsyska — APPROVED** [review]

**Review by CreativePhilip — CHANGES_REQUESTED** [review]

File: rms/__main__.py (hunk 1 above)
Comment (jankrokos) [reply rms/__main__.py]: fixed

File: rms/auth/dependencies.py (hunk 2 above)
Comment (jankrokos) [reply rms/auth/dependencies.py]: fixed

**Review by CreativePhilip — APPROVED** [review]

**Pull request merged**